forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/BiometricAndDeviceCredential.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023 ForgeRock. All rights reserved.
+ *
+ *  This software may be modified and distributed under the terms
+ *  of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth.devicebind
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Build
+import android.security.keystore.KeyProperties
+import androidx.biometric.BiometricManager
+import androidx.biometric.BiometricPrompt
+import org.forgerock.android.auth.callback.Attestation
+import org.forgerock.android.auth.callback.DeviceBindingAuthenticationType
+import java.security.PrivateKey
+import java.security.interfaces.RSAPublicKey
+
+/**
+ * Settings for all the biometric authentication and device credential is configured
+ */
+open class BiometricAndDeviceCredential : BiometricAuthenticator() {
+    override fun authenticate(authenticationCallback: BiometricPrompt.AuthenticationCallback,
+                              privateKey: PrivateKey) {
+        biometricInterface.authenticate(authenticationCallback)
+    }
+
+    /**
+     * generate the public and private keypair
+     */
+    @SuppressLint("NewApi")
+    override suspend fun generateKeys(context: Context, attestation: Attestation): KeyPair {
+        val builder = cryptoKey.keyBuilder()
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            builder.setAttestationChallenge(attestation.challenge)
+        }
+
+        //We use time-base key because we allow device credential as fallback
+        //Device credential is not consider biometric strong
+        if (isApi30OrAbove) {
+            builder.setUserAuthenticationParameters(cryptoKey.timeout,
+                KeyProperties.AUTH_BIOMETRIC_STRONG or KeyProperties.AUTH_DEVICE_CREDENTIAL)
+        } else {
+            builder.setUserAuthenticationValidityDurationSeconds(cryptoKey.timeout)
+        }
```

Comment:
For BiometricAndDeviceCredential, we keep using the time-based key, because the Device Credential is not consider strong Biometric.

Reply:
yes this make sense ,. should we update the docs mentioning what keys are used by different authenticator type? 

for example: BiometricOnly uses authBased , BiometricAndDeviceCredential uses timebased

Reply:
For now, I don't think it is necessary to get to that detail.